File: android/core/src/main/java/uk/co/amlcurran/social/EventCalendarItem.java
package uk.co.amlcurran.social;

import javax.annotation.Nonnull;

class EventCalendarItem implements CalendarItem {
    private final String eventId;
    private final String title;
    private final Timestamp start;
    private final Timestamp endTime;

    public EventCalendarItem(String eventId, String title, Timestamp time, Timestamp endTime) {
        this.eventId = eventId;
        this.title = title;
        this.start = time;
        this.endTime = endTime;
    }

    public String id() {
        return eventId;
    }

    @Nonnull
    @Override
    public String title() {
        return title;
    }

    @Nonnull
    @Override
    public Timestamp startTime() {
        return start;
    }

    @Nonnull
    @Override
    public Timestamp endTime() {
        return endTime;
    }

    @Override
    public boolean isEmpty() {
        return false;
    }
}

File: android/core/src/main/java/uk/co/amlcurran/social/EventCalendarItem.kt
package uk.co.amlcurran.social

class EventCalendarItem(private val eventId: String, private val title: String, private val start: Timestamp, private val endTime: Timestamp) : CalendarItem {

    fun id(): String {
        return eventId
    }

    override fun title(): String {
        return title
    }

    override fun startTime(): Timestamp {
        return start
    }

    override fun endTime(): Timestamp {
        return endTime
    }

    override var isEmpty: Boolean = false
}
